document-portal/document-portal.c

```diff
@@ -1416,6 +1416,68 @@ portal_list (GDBusMethodInvocation *invocation,
   return TRUE;
 }
 
+static gboolean
+portal_get_real_path (GDBusMethodInvocation *invocation,
+             GVariant *parameters,
+             XdpAppInfo *app_info)
+{
+  const char *id = NULL;
+  g_autoptr(PermissionDbEntry) entry = NULL;
+
+  g_variant_get (parameters, "(&s)", &id);
+
+  XDP_AUTOLOCK (db);
+
+  entry = permission_db_lookup (db, id);
+
+  if (!entry)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                                             "Invalid ID passed");
+      return TRUE;
+    }
+
+  if (!xdp_app_info_is_host (app_info))
+    {
+      g_autofree const char **apps = NULL;
+      g_autofree const char *app_id = NULL;
+      static gboolean id_found = FALSE;
```

Comment:
Why is this variable static? Wouldn't it make it always return true in case we already successfully found one?